src/firefly/js/api/ApiViewer.js

```diff
@@ -117,6 +117,19 @@ export function getViewer(channel,file=defaultViewerFile) {
     }
 }
 
+/**
+ * wrapper function to return the API's getViewer object.  This allow one firefly app to
+ * gain access to another app's API.  To use this function, it should be loaded first.  @see ApiViewer#loadRemoteApi
+ * @param {object} p            parmeter
+ * @param {String} [p.channel]  default to current connected channel.
+ * @param {String} [p.file]     url path, or the html file to load.  Defaults to blank(index of the app).
+ * @param {String} [p.iframeId]
+ */
+export function getRemoteViewer({channel, file, iframeId}) {
+    const win = iframeId ? document.getElementById(iframeId).contentWindow : window;
+    const getViewer = get(win, 'firefly.getViewer');
+    return getViewer && getViewer(channel, file);
+}
```

Comment:
A few comments:

- Where is loadRemoteApi?  Has it been written yet?
- What is the difference between the name getViewer and getRemoteViewer? They are both really the same.  They both want to talk to a viewer
- right now, `getRemoteViewer` will act like `getViewer` if channel and file are only passed.
- could the iframe id and the channel id be the same?  In other words, we always name an iframe by this new channel we a using to talk? They we would not need a iframeId parameter. It could check using channel.
- Down the road if we removed the iframe we could not touch the api.
- I do like the fact that the rest of the codes does not need to know it is talking to an iframe 
- I think:
    - `getRemoteViewer` should be merged with `getViewer` and called `getViewer`
    -  take 2 parameters as it does now
     - have coded added to check it an iframe with the channel name exist
     - it it does, then operate as `getRemoveViewer` otherwise operate as `getViewer`

Reply:
IMO, this is how getViewer should be:
`getViewer({channel, file, baseUrl})`
- baseUrl is where you can locate firefly_loader.js, i.e /irsaviewer, /applications/finderchart, etc
  - use baseUrl as the frame-id, if iframe is used.
  - we should not tie channel to baseUrl because it's possible to have multiple channels. 
- it should be async, either via promise(preferred but not fully supported) or a callback.
  - this allow for lazy-loading and does not require loading the api separately.
- the single parameter with properties is to allow for more parameters in the future without fear of breaking the api.

If we want to do the above and still be backward compatible, we can make `getRemoteViewer` the new api and deprecate `getViewer`.  `getViewer` will call `getRemoteViewer` with immediate return for a `synch` behavior.

Reply:
I guess you could do that.  That might even work better if the future if we make part of it npm installable.  But, you could since you could still use `getViewer`, If it is passed an object just use the new behavior, if a string use the old.

Reply:
promises are support if we load firefly because we  bring them in.

Reply:
OK, I am having second thoughts about `getViewer()` returning a promise. I think your original two function approach is better.  The reason is that for 90% of the use cases it makes `getViewer()` more complex. I think we need to make it simple for the normal use case.